src/core/gui/PagePreviewDecoration.cpp

```diff
@@ -0,0 +1,99 @@
+#include "PagePreviewDecoration.h"
+
+#include <math.h>
+
+#include "../control/Control.h"
+#include "../control/settings/Settings.h"
+
+void PagePreviewDecoration::drawDecoration(cairo_t* cr, SidebarPreviewPageEntry* pageEntry, Control* control) {
+    switch ( control->getSettings()->getSidebarNumberingStyle()) {
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_1:
+        PagePreviewDecoration::drawStyle1(cr, pageEntry, control);
+        break;
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_2:
+        PagePreviewDecoration::drawStyle2(cr, pageEntry, control);
+        break;
+    case SidebarNumberingStyle::SIDEBAR_NUM_STYLE_3:
+        PagePreviewDecoration::drawStyle3(cr, pageEntry, control);
+        break;
+    default:
+        break;
+    }
+
+}
+
+void PagePreviewDecoration::drawStyle1(cairo_t* cr, SidebarPreviewPageEntry* pageEntry, Control* control) {
+    cairo_text_extents_t extents;
```

Comment:
In general, whenever drawing something using cairo and modifying the state of the cairo context (`cairo_t`), it's good to have a `cairo_save`/`cairo_restore` pair to ensure we are not altering the state for other functions.

We have a RAII wrapper doing just that:
```suggestion
    xoj::util::CairoSaveGuard saveGuard(cr);
    cairo_text_extents_t extents;
```
This will call `cairo_save(cr)` upon construction, and `cairo_restore(cr)` upon destruction (at the end of the function), so we never forget to `cairo_restore`.

Reply:
Didn't know that about cairo, now I have added the save guard. See: [[4be6861](https://github.com/xournalpp/xournalpp/pull/4693/commits/4be68616940e59af1b6d57b60d60dcdab3316a57)]

Should such a save guard also be added to the SidebarPreviewBaseEntry::paint() method?